### PR TITLE
Workaround for ilab script source

### PR DESCRIPTION
**Pull request by enriquebelarte**
As [this](https://github.com/containers/ai-lab-recipes/pull/599) PR threw an [error](https://github.com/containers/ai-lab-recipes/pull/606) that prevented the build process, I propose this workaround that would make Konflux CI build pipeline work but also won't change the `make` behavior.

Suggestions, corrections accepted.
@n1hility @Gregory-Pereira

### CODE REVIEW

**Review by lmilbaum — CHANGES_REQUESTED** [review]
Please sign-off to pass the DCO check

File: training/nvidia-bootc/add_ilab_wrapper.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+LOCAL_ILAB_WRAPPER="run/.input/ilab"
+
+if [ -f "$LOCAL_ILAB_WRAPPER" ]; then
+	cp "$LOCAL_ILAB_WRAPPER" /usr/local/bin/ilab
+else
+	curl -o /usr/local/bin/ilab "https://raw.githubusercontent.com/enriquebelarte/ai-lab-recipes-upstream/main/training/ilab-wrapper/ilab"
```
Comment (lmilbaum) [inline training/nvidia-bootc/add_ilab_wrapper.sh]: Can you please move the ilab script/binary to a org repo?

File: training/nvidia-bootc/Containerfile [hunk 2]
```diff
@@ -99,6 +99,7 @@ COPY --from=builder /home/builder/yum-packaging-precompiled-kmod/RPMS/*/*.rpm /r
 COPY --from=builder --chmod=444 /home/builder/yum-packaging-precompiled-kmod/tmp/firmware/*.bin /lib/firmware/nvidia/${DRIVER_VERSION}/
 # Temporary workaround until the permanent fix for libdnf is merged
 COPY nvidia-toolkit-firstboot.service /usr/lib/systemd/system/nvidia-toolkit-firstboot.service
+COPY add_ilab_wrapper.sh add_ilab_wrapper.sh
```
Comment (rhatdan) [inline training/nvidia-bootc/Containerfile]: Could you do the script below as inside of the run command?
```
if [ -f /run/.input/ilab" ]; then \
	cp /run/.input/ilab /usr/local/bin/ilab \
else \
	curl -o /usr/local/bin/ilab "https://raw.githubusercontent.com/containers/ai-lab-recipes-upstream/main/training/ilab-wrapper/ilab" \
fi
```

Comment (enriquebelarte) [reply training/nvidia-bootc/Containerfile]: Sure. Much better. I'm updating it.

File: training/nvidia-bootc/Containerfile [hunk 3]
```diff
@@ -157,7 +157,11 @@ fi
 RUN grep -q /usr/lib/containers/storage /etc/containers/storage.conf || \
     sed -i -e '/additionalimage.*/a "/usr/lib/containers/storage",' \
 	/etc/containers/storage.conf && \
-	cp /run/.input/ilab /usr/local/bin/ilab
+	if [ -f /run/.input/ilab" ]; then \
```
Comment (Gregory-Pereira) [inline training/nvidia-bootc/Containerfile]: Missing a set of quotations:
```suggestion
	if [ -f "/run/.input/ilab" ]; then \
```

File: training/nvidia-bootc/Containerfile [hunk 4]
```diff
@@ -157,7 +157,11 @@ fi
 RUN grep -q /usr/lib/containers/storage /etc/containers/storage.conf || \
     sed -i -e '/additionalimage.*/a "/usr/lib/containers/storage",' \
 	/etc/containers/storage.conf && \
-	cp /run/.input/ilab /usr/local/bin/ilab
+	if [ -f "/run/.input/ilab" ]; then \
+	  cp /run/.input/ilab /usr/local/bin/ilab \
+        else \
+	  curl -o /usr/local/bin/ilab "https://raw.githubusercontent.com/containers/ai-lab-recipes-upstream/main/training/ilab-wrapper/ilab" \
```
Comment (lmilbaum) [inline training/nvidia-bootc/Containerfile]: Is containers/ai-lab-recipes-upstream a new repo?

Comment (Gregory-Pereira) [reply training/nvidia-bootc/Containerfile]: Excellent catch, it is not.

File: training/nvidia-bootc/add_ilab_wrapper.sh (hunk 1 above)
Comment (lmilbaum) [reply training/nvidia-bootc/add_ilab_wrapper.sh]: Does that mean that this change is not tested in CI?

Comment (enriquebelarte) [reply training/nvidia-bootc/add_ilab_wrapper.sh]: Sorry, just a typo

**Comment by Gregory-Pereira** [comment]
As for this not triggering CI I will take a look now.

**Comment by Gregory-Pereira** [comment]
I think tests will only run if this was an upstream branch for access to secrets.

**Comment by rhatdan** [comment]
LGTM

**Comment by Gregory-Pereira** [comment]
Ill fix CI in a follow on, if Dan approves we can merge now.

**Review by rhatdan — APPROVED** [review]
LGTM

**Pull request merged**